trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1017,6 +1147,28 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
         }
     }
 
+    /// Advances a query (if one is active for the node) using the received ENRs.
+    /// Does nothing if called with a node_id that does not have a corresponding active query request.
+    fn advance_query(&mut self, source: Enr, enrs: Vec<Enr>, query_id: QueryId) {
```

Comment:
What do you think about changing the function signature to have a `Query` parameter instead of `QueryId`? With this change, it would not be possible to call for a query unless that query exists in the map.

Reply:
I tried this. Taking the query directly in the function signature required calling out the specific types of the query:
`fn advance_query(&mut self, source: Enr, enrs: Vec<Enr>, query: Query<QueryInfo, NodeID, Enr>) ` whereas the current approach allows everything to stay generic, which I find cleaner. 

I also prefer checking whether the `QueryId` maps to a valid `Query` within the query-specific function, rather than having that logic bloat the `process_nodes` function.